src/uct/ib/base/ib_iface.c

```diff
@@ -276,7 +277,9 @@ void uct_ib_address_pack(const union ibv_gid *gid, uint16_t lid,
         *(uint16_t*) ptr = lid;
         ptr              = UCS_PTR_BYTE_OFFSET(ptr, sizeof(uint16_t));
 
-        if (is_global_addr) {
+        /* if is_global_addr is false, *gid is undefined */
+        if (is_global_addr ||
+            (gid->global.subnet_prefix != UCT_IB_LINK_LOCAL_PREFIX)) {
```

Comment:
we can't access gid if `is_global_addr` is `false`.
would need to remove the second check. We always want to pack iface_id if address is global, right?

Reply:
right... fixed

Reply:
not correct, in SRIOV forcing global_addr will not work now

Reply:
then we need additional flag here: 1-st one for GID available, second - forced GRH... are we ok with this?

Reply:
why not just revert to passing iface->is_global_addr from uct_ib_iface_address_pack?
why did you have to change it to passing 1 always?

Reply:
because original condition was
```
if ((gid->global.subnet_prefix != UCT_IB_LINK_LOCAL_PREFIX) || is_global_addr) {...
```
which may crash in case if CM pass ```is_global_addr==0``` because gid may point into unavailable memory...
but I think I found solution